mu-plugins/central-hub/tests/phpunit/unit/config-store/getAllKeys.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ *  Tests for getAllKeys()
+ *
+ * @package    spiralWebDb\centralHub\Tests\Unit\ConfigStore
+ * @since      1.3.0
+ * @author     Robert A. Gadon
+ * @link       https://github.com/rgadon107/cornerstone
+ * @license    GNU General Public License 2.0+
+ */
+
+namespace spiralWebDb\centralHub\Tests\Unit\ConfigStore;
+
+use Brain\Monkey;
+use function KnowTheCode\ConfigStore\getAllKeys;
+use spiralWebDb\Cornerstone\Tests\Unit\Test_Case;
+
+/**
+ * Class Tests_GetAllKeys
+ *
+ * @package spiralWebDb\centralHub\Tests\Unit\ConfigStore
+ * @group   config-store
+ */
+class Tests_GetAllKeys extends Test_Case {
+
+	/**
+	 * Prepares the test environment before each test.
+	 */
+	protected function setUp() {
+		parent::setUp();
+
+		require_once CENTRAL_HUB_ROOT_DIR . '/src/config-store/api.php';
+	}
+
+	/**
+	 * Test getAllKeys() should return the array keys from `_the_store()`.
+	 */
+	public function test_should_return_the_array_keys_from_the_store() {
+		Monkey\Functions\expect( 'array_keys' )
```

Comment:
@rgadon107 Never mock native PHP constructs such as `array_keys`.  These are built into the language.  We know the expected behavior.

But look at this function in the API:

https://github.com/rgadon107/cornerstone/blob/866c93da11429abf56c811148596f539b08e1e12/mu-plugins/central-hub/src/config-store/api.php#L103-L107

What internal dependent function is it using?  `_the_store()`.  You should be mocking the `_the_store()` and then validating what you get back when requesting all of the keys. Hint Hint.

Reply:
@hellofromtonya Tried this. See my [latest commit ](https://github.com/rgadon107/cornerstone/pull/24/commits/6bc980830996a761ab3f75dde1b31bbbce53ccbc). Keep getting an error from `array_keys()` with the API function `getAllKeys`. Also, see my comment below for details.